gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -212,14 +215,17 @@ else if (statement instanceof ExecuteImmediate executeImmediate) {
         }
         catch (IOException e) {
             log.warn("Error extracting request body for rules processing: %s", e.getMessage());
+            parseErrorMessage = Optional.ofNullable(e.getMessage());
```

Comment:
This isn't a parse error because IOException happens before parsing. The variable should be renamed or we shouldn't set the variable here.